mycore-neo4j/src/main/java/org/mycore/mcr/neo4j/utils/MCRNeo4JQueryRunner.java

```diff
@@ -0,0 +1,235 @@
+            List<Map<String, String>> records = new ArrayList<>();
+            Result result = tx.run(queryString);
+            while (result.hasNext()) {
+               Record thisRecord = result.next();
+               Map<String, String> keyMap = new HashMap<>();
+
+               for (String key : thisRecord.keys()) {
+                  Value recordData = thisRecord.get(key);
+                  counter.getAndAdd(1);
+                  if (StringUtils.equals(recordData.type().name(), "NODE")) {
+                     String node = nodeToJson(recordData.asNode(), gson, lang);
+                     String nodeSB = "{\"n\":" + node + "}";
+                     LOGGER.debug("record is Node");
+                     keyMap.put("node_" + key, nodeSB);
+
+                  } else if (StringUtils.equals(recordData.type().name(), "RELATIONSHIP")) {
+                     LOGGER.debug("record is Relationship");
+                     keyMap.put("rel_" + key, "{\"r\":" + gson.toJson(recordData.asRelationship()) + "}");
+                  } else if (StringUtils.equals(recordData.type().name(), "PATH")) {
+                     StringBuilder pathSB = new StringBuilder();
+                     LOGGER.debug("record is Path");
+                     //Gather Stuff
+                     Path neo4jPath = recordData.asPath();
+                     Node startNode = neo4jPath.start();
+                     Node endNode = neo4jPath.end();
+                     Iterable<Relationship> relationships = neo4jPath.relationships();
+
+                     // Parse Stuff to use full Json
+                     String start = nodeToJson(startNode, gson, lang);
+                     String end = nodeToJson(endNode, gson, lang);
+                     String relationshipJson = gson.toJson(relationships);
+
+                     pathSB.append("{\"p\":{\"nodes\":[");
+                     pathSB.append(start).append(',');
+
+                     pathSB.append(end).append("],");
+                     pathSB.append("\"relationships\":").append(relationshipJson);
+
+                     pathSB.append("}}");
+
+                     keyMap.put("path_" + key, pathSB.toString());
+                  } else if (StringUtils.equals(recordData.type().name(), "NULL")) {
+                     LOGGER.warn("Got record of type {} for key {} which is not parsed and is ignored",
+                           recordData.type().name(), key);
+                  } else {
+                     LOGGER.warn("Got record of type {} for key {} which is not parsed",
+                           recordData.type().name(), key);
+                     keyMap.put(key, gson.toJson(thisRecord.asMap()));
+                  }
+               }
+               records.add(keyMap);
+
+            }
+            return records;
+         });
+      }
+
+   }
+
+   private static String nodeToJson(Node startNode, Gson gson, String lang) {
+      String elementId = startNode.elementId();
```

Comment:
What's happening here
a) create Json with GSON
b) serialize to String with GSON
c) create Neo4JNodeJsonRecord by parsing the JSON-String via ObjectMapper
d) serialize the object again to String via ObjectMapper

Why don't you simply create the Neio4NOdeJsonRecord as Java-POJO?

Reply:
cleaned the function by building the POJO